app/models/user.rb

```diff
@@ -23,4 +23,18 @@ def digest string
   def self.new_token
     SecureRandom.urlsafe_base64
   end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
```

Comment:
what is the difference between  update_attribute and update_attributes

Reply:
Hàm update_attribute chỉ update 1 record, bỏ qua tất cả validate trong model user.
Hàm update_attributes thì update được nhiều attribute (tất cả attribute được truyền vào từ params của object đó) nhưng chỉ khi pass validate thì mới thực hiện update.